File: reverse-hash.py
def hash(s):
	h = 7
	letters = 'acdegilmnoprstuw'
	for i in range(len(s)):
		h = (h*37 + letters.index(s[i]))

	return h

def solution(hash_value):
	letters = 'acdegilmnoprstuw'
	res = ''
	while hash_value > 0:
		remainder = hash_value % 37
		if remainder <= len(letters):
			res += letters[remainder]
			hash_value = (hash_value - remainder) // 37
		else:
			return

		if hash_value == 7:
			return res[::-1]
	
	return None	

print(solution(680131659347))
print(solution(930846109532517))

File: reverse-hash.js
function solution(hash_value) {
	const letters = 'acdegilmnoprstuw';
	let res = [];
	while (hash_value > 0) {
		remainder = hash_value % 37;
		if (remainder <= letters.length){
			res.push(letters[remainder]);
			hash_value = Math.floor((hash_value - remainder)/37)
		} else
			return null;

		if (hash_value === 7) return res.reverse().join('');
	}
	return null;
}

console.log(solution(680131659347));
console.log(solution(930846109532517))